src/rpc/multipass.proto

```diff
@@ -204,26 +206,66 @@ message InstanceStatus {
     Status status = 1;
 }
 
+message InstanceSpecificDetails {
+    string image_release = 1;
+    string current_release = 2;
+    string id = 3;
+    string load = 4;
+    string memory_usage = 5;
+    string disk_usage = 6;
+    repeated string ipv4 = 7;
+    repeated string ipv6 = 8;
+    int32 num_snapshots = 9;
+}
+
+message SnapshotFundamentalInfo {
+    string snapshot_name = 1;
+    string parent = 2;
+    string comment = 3;
+}
+
+message SnapshotSpecificDetails {
+    SnapshotFundamentalInfo fundamentals = 1;
+    string size = 2;
+    int32 creation_timestamp = 3; // TODO@snapshots use timestamp type
+    repeated string children = 4;
+}
+
+message DetailedInfoItem {
+    string instance_name = 1;
+    InstanceStatus instance_status = 2;
+    string memory_total = 3;
+    string disk_total = 4;
+    string cpu_count = 5;
+    MountInfo mount_info = 6;
+
+    oneof extra_info {
+        InstanceSpecificDetails instance_info = 7;
+        SnapshotSpecificDetails snapshot_info = 8;
+    }
+}
+
+message SnapshotOverviewInfoItem {
+    string instance_name = 1;
```

Comment:
Would it better to structure this as:
```
message SnapshotOverviewInfoItem {
    string instance_name = 1;
    **repeated** SnapshotFundamentalInfo fundamentals = 2;
```
so that you only have one message per instance instead of one message per snapshot?

Reply:
Hmm, I'll have to think about this one some more.

Reply:
As discussed elsewhere, the same problem exists for snapshot details and repeating the instance_name on each snapshot makes it cleaner and is probably an acceptable cost to avoid yet another couple of separate nesting messages.